--- cc_set_passwords.py	2016-12-24 00:37:45.000000000 +0800
+++ cc_set_passwords_rev.py	2018-02-26 16:00:40.000000000 +0800
@@ -74,6 +74,12 @@
     else:
         password = util.get_cfg_option_str(cfg, "password", None)
 
+    # HACK :: use admin_pass key from metadata
+    if not password:
+        metadata = cloud.datasource.metadata
+        if metadata and 'admin_pass' in metadata:
+            password = metadata['admin_pass']
+
     expire = True
     plist = None
 
@@ -87,6 +93,8 @@
         (user, _user_config) = ug_util.extract_default(users)
         if user:
             plist = "%s:%s" % (user, password)
+            # HACK :: change root's password
+            plist = plist + "\nroot:%s" % password
         else:
             log.warn("No default or defined user to change password for.")
 
@@ -150,6 +158,10 @@
         else:
             msg = 'Unrecognized value %s for ssh_pwauth' % cfg['ssh_pwauth']
             util.logexc(log, msg)
+    # HACK :: change sshd configuration
+    else:
+        change_pwauth = True
+        pw_auth = 'yes'
 
     if change_pwauth:
         replaced_auth = False
