--- cc_set_passwords.py	2018-02-13 02:20:26.541020193 +0000
+++ cc_set_passwords_rev.py	2018-02-13 02:23:38.741449208 +0000
@@ -74,6 +74,12 @@
     else:
         password = util.get_cfg_option_str(cfg, "password", None)
 
+    # HACK :: use admin_pass key from metadata
+    if not password:
+        metadata = cloud.datasource.metadata
+        if metadata and 'admin_pass' in metadata:
+            password = metadata['admin_pass']
+
     expire = True
     plist = None
 
@@ -87,6 +93,8 @@
         (user, _user_config) = ug_util.extract_default(users)
         if user:
             plist = "%s:%s" % (user, password)
+            # HACK :: change root's password
+            plist = plist + "\nroot:%s" % password
         else:
             log.warn("No default or defined user to change password for.")
 
